### PR TITLE
C7 0.4 support and API update

### PR DESCRIPTION
**Pull request by dreymonde**

### CODE REVIEW

File: Sources/FileStream.swift [hunk 1]
```diff
@@ -43,37 +43,41 @@ public final class FileStream: Stream {
...
-    public func receive() throws -> Data {
+    public func receive(upTo byteCount: Int, timingOut deadline: Double = .never) throws -> Data {
         try assertNotClosed()
         do {
             if file.eof {
                 file.close()
                 return []
             } else {
-                return try file.read(lowWaterMark: lowWaterMark, highWaterMark: highWaterMark)
+                return try file.read(lowWaterMark: lowWaterMark, highWaterMark: byteCount, timingOut: deadline)
```
Comment (goloveychuk) [inline Sources/FileStream.swift]: maybe there should be `min(byteCount, hightWaterMark)`?

**Comment by paulofaria** [comment]
`File` should follow the approach used in `TCP`. https://github.com/VeniceX/TCP/blob/master/Source/TCPConnection.swift#L122

File: Sources/FileStream.swift (hunk 1 above)
Comment (dreymonde) [inline Sources/FileStream.swift]: Yeah, that should work. So the user will be always sure that nothing will overcome `highWaterMark`.

**Comment by dreymonde** [comment]
@paulofaria you mean having the different methods which takes `Duration` or `Double`?

**Comment by paulofaria** [comment]
I think we still need:

``` swift
public var standardInputStream = try! File(fileDescriptor: STDIN_FILENO)
public var standardOutputStream = try! File(fileDescriptor: STDOUT_FILENO)
public var standardErrorStream = try! File(fileDescriptor: STDERR_FILENO)
```

**Comment by paulofaria** [comment]
really not sure about `File` having `send` and `write`. could be confusing.

**Pull request merged**